desktop_native/src/password/windows.rs

```diff
@@ -0,0 +1,158 @@
+use anyhow::{anyhow, Result};
+use widestring::{U16CString, U16String};
+use windows::Win32::{
+    Foundation::{GetLastError, FILETIME, PWSTR},
+    Security::Credentials::{
+        CredDeleteW, CredFree, CredReadW, CredWriteW, CREDENTIALW, CRED_FLAGS,
+        CRED_PERSIST_ENTERPRISE, CRED_TYPE_GENERIC,
+    },
+};
+
+const CRED_FLAGS_NONE: u32 = 0;
+
+pub async fn get_password<'a>(service: &str, account: &str) -> Result<String> {
+    let target_name = U16CString::from_str(target_name(service, account))?;
+
+    let mut credential: *mut CREDENTIALW = std::ptr::null_mut();
+    let credential_ptr = &mut credential;
+
+    let result = unsafe {
+        CredReadW(
+            PWSTR(target_name.as_ptr()),
+            CRED_TYPE_GENERIC.0,
+            CRED_FLAGS_NONE,
+            credential_ptr,
+        )
+    };
+
+    scopeguard::defer!({
+        unsafe { CredFree(credential as *mut _) };
```

Comment:
It seems unlikely the crate is designed to be used with so much `unsafe`.

Is the issue that no wrapper functions exist for things like `CredFree` or `GetError`? I'm worried we're using the library wrong...

Reply:
Sadly it's supposed to be used that way. The samples all contain big chunks of `unsafe`.